File: src/main/kotlin/com/vishnu/DBAccess.java
package com.vishnu;

import java.sql.*;
import java.text.SimpleDateFormat;

public class DBAccess {

    public static void main(String[] args) throws SQLException {
        DBAccess.connectToAndQueryDatabase(args[0], args[1], args[2]);
    }

    private static void connectToAndQueryDatabase(String username, String password, String url) throws SQLException {

        System.out.println("Connecting " + DBAccess.getDateTime());

        Connection con = DriverManager.getConnection(url, username, password);

        System.out.println("Connected " + DBAccess.getDateTime());

        execute("SELECT * FROM SELR_USR_PRFL", con);
        execute("SELECT j.id, j.name FROM APPL_CONFIG A, JSON_TABLE( A.CONFIG_JSON, '$.*.config[*]'COLUMNS(name PATH '$.Name' , id PATH '$.ID')) j WHERE CONFIG_KEY <> 'DC'", con);

    }

    private static void execute(String query, Connection con) throws SQLException {
        Statement stmt = con.createStatement();
        ResultSet rs = stmt.executeQuery(query);

        System.out.println("Executed query " + DBAccess.getDateTime());

        int count = 0;
        while (rs.next()) {
            count++;
        }

        System.out.println("read rows " + count + " " + DBAccess.getDateTime());
    }

    private static String getDateTime() {
        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd 'at' HH:mm:ss.SSS z");
        Date date = new Date(System.currentTimeMillis());
        return (formatter.format(date));
    }
}

File: src/main/kotlin/com/vishnu/DBAccess.kt
package com.vishnu

import java.sql.Connection
import java.sql.Date
import java.sql.DriverManager
import java.sql.SQLException
import java.text.SimpleDateFormat

object DBAccess {

    private val dateTime: String
        get() {
            val formatter = SimpleDateFormat("yyyy-MM-dd 'at' HH:mm:ss.SSS z")
            val date = Date(System.currentTimeMillis())
            return formatter.format(date)
        }

    @Throws(SQLException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        DBAccess.connectToAndQueryDatabase(args[0], args[1], args[2])
    }

    @Throws(SQLException::class)
    private fun connectToAndQueryDatabase(username: String, password: String, url: String) {

        println("Connecting ${DBAccess.dateTime}")
        val con = DriverManager.getConnection(url, username, password)
        println("Connected ${DBAccess.dateTime}")

        execute(
            query = "SELECT * FROM SELR_USR_PRFL",
            con = con
        )
        execute(
            query = "SELECT j.id, j.name FROM APPL_CONFIG A, JSON_TABLE( A.CONFIG_JSON, '$.*.config[*]'COLUMNS(name PATH '$.Name' , id PATH '$.ID')) j WHERE CONFIG_KEY <> 'DC'",
            con = con
        )

    }

    @Throws(SQLException::class)
    private fun execute(query: String, con: Connection) {
        val stmt = con.createStatement()
        println("Current fetchSize ${stmt.fetchSize}")
        stmt.fetchSize = 100
        println("New fetchSize ${stmt.fetchSize}")
        val rs = stmt.executeQuery(query)

        println("Executed query ${DBAccess.dateTime}")

        var count = 0
        while (rs.next()) {
            count++
        }

        println("read rows $count ${DBAccess.dateTime}")
    }
}
